src/guides/components/endpoint.md

```diff
@@ -0,0 +1,253 @@
+
+## sdk.api.register()
+
+This method allows you to register functions that can be called from the frontend. Meaning you can link a frontend action, like a button click, to the execution of some code.
+
+### Registering an API Endpoint
+
+For example, the following could be the contents of your plugin's `/packages/backend/src/index.ts` file:
+
+First, the necessary type aliases are imported. `SDK` is the interface used to interact with Caido. `DefineAPI` is used to structure the API: definining what methods or endpoints are available, the parameters those methods accept and what types of values they return.
+
+``` ts
+import { SDK, DefineAPI } from "caido:plugin";
+```
+
+Next, the function is defined. The function takes three parameters: `sdk`, `a` and `b`. The `sdk` parameter is typed using the `SDK` alias to give the function access to its utilities. The `a` and `b` parameters are expected to be numbers as this function multiplies the two together.
+
+``` ts
+function multiply(sdk: SDK, a: number, b: number) {
+    const result = a * b;
+    sdk.console.log(`The product of the multiply call is: ${result}`);
+    return result;
+}
+```
+
+Using the `DefineAPI` utility, we are stating what our API offers. In this case, the `multiply` function is available to be called. To ensure the function receives the expected parameter data types, `typeof` is used to link the `multiply` API call to the `multiply` function definition. This definition is stored in the type alias `API` and exported so it can be used in other files.
+
+``` ts
+export type API = DefineAPI<{
+    multiply: typeof multiply;
+}>;
+```
+
+Next, we define a function that will run as soon as Caido loads the plugin. It extends upon the base `SDK` by adding the `<API>`. In order to register the function, we use the `sdk.api.register()` method which takes two parameters: a string name for the function and the function it refers to. We give the name `"multiply"` to the `multiply` function.
+
+``` ts
+export function init(sdk: SDK<API>) {
+    sdk.api.register("multiply", multiply);
+}
+```
+
+::: tip
+To view the entire backend script, expand the following:
+
+<details>
+<summary>Example</summary>
+
+``` ts
+import { SDK, DefineAPI } from "caido:plugin";
+
+function multiply(sdk: SDK, a: number, b: number) {
+    const result = a * b;
+    sdk.console.log(`The product of the multiply call is: ${result}`);
+    return result;
+}
+
+export type API = DefineAPI<{
+    multiply: typeof multiply;
+}>;
+
+export function init(sdk: SDK<API>) {
+    sdk.api.register("multiply", multiply);
+}
+```
+
+</details>
+:::
+
+## sdk.backend.multiply()
+
+This method allows you to call the functions that have been registered and defined in your API. The given name of the function is appended to `sdk.backend.`.
+
+### Calling the API Endpoint
+
+---
+
+<img alt="Calling the API." src="/_images/api_register_function.png" center/>
```

Comment:
I think this is a good screenshot to show at the end of the article.

The outline of the article ends up being:
- "Here's what we're gonna do"
- "Here's how to do it"
- "Here's what the final result looks like"